pyanaconda/modules/subscription/subscription.py

```diff
@@ -174,3 +252,102 @@ def set_system_purpose_data(self, system_purpose_data):
+        """
+        # process the request from the "outside" to make it is valid
+        # before setting it to the module
+        self._subscription_request = self._process_subscription_request(
+            subscription_request
+        )
+        self.subscription_request_changed.emit()
+        log.debug("A subscription request has been set.")
+
+    def _process_subscription_request(self, subscription_request):
+        """Process a subscription request and make sure it is valid.
+
+        Make sure a subscription request coming from the outside
+        is in a valid state before it is set into the module:
+        - make sure the properties indicating if a private field
+          is set are valid
+        - prevent blank private fields from overriding previously
+          set private values
+        """
+        current_request = self._subscription_request
+
+        # set private data indicators
+        subscription_request.account_password_set = bool(
+                subscription_request.account_password
+        )
+        subscription_request.activation_keys_set = bool(
+                subscription_request.activation_keys
+        )
+        subscription_request.server_proxy_password_set = bool(
+                subscription_request.server_proxy_password
+        )
+
+        # Prevent currently set private data from being wiped if the
+        # new structure has them set to an empty string.
+        # - if an incoming password is an empty string &
+        #   a non empty string is already set as password,
+        #   keep the currently set password
+        # - if an incoming activation key list is empty
+        #   but we have a non-empty list set, keep the
+        #   currently set list of activation keys
+        if subscription_request.account_password == "":
+            subscription_request.account_password = current_request.account_password
+        if subscription_request.activation_keys == []:
+            subscription_request.activation_keys = current_request.activation_keys
+        if subscription_request.server_proxy_password == "":
+            subscription_request.server_proxy_password = current_request.server_proxy_password
+
+        return subscription_request
+
+    @property
+    def connect_to_insights(self):
+        """Indicates if the target system should be connected to Red Hat Insights.
+
+        :return: True to connect, False not to connect the target system to Insights
+        :rtype: bool
+        """
+        return self._connect_to_insights
+
+    def set_connect_to_insights(self, connect):
+        """Set if the target system should be connected to Red Hat Insights.
+
+        :param bool connect: set to True to connect, set to False not to connect
+        """
+        self._connect_to_insights = connect
+        self.connect_to_insights_changed.emit()
+        log.debug("Connect target system to Insights set to: %s", self._connect_to_insights)
```

Comment:
This should have been in a separate commit. Please try to split the code to multiple commits if possible next time.

Reply:
Given that the Insights hint is manipulated by the rhsm command via the `--connect-to-insights` parameter I think it makes sense to include it in the same commit that includes the SubscriptionRequest structure which is mainly filled by data from the same kickstart command. Same thing for tests that test that the rhsm command is parsed correctly.

Reply:
It's not problem to add something dependent in separate commits if you won't broke the tests during the way.